File: apk_monytix/app/src/main/java/com/example/apk/console/tabs/OverviewTabView.kt
package com.example.apk.console.tabs

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.example.apk.console.MolyConsoleViewModel
import com.example.apk.ui.components.GlassCard
import com.example.apk.ui.theme.Charcoal
import com.example.apk.ui.theme.Gold
import com.example.apk.ui.theme.TextPrimary
import com.example.apk.ui.theme.TextSecondary
import com.example.apk.console.utils.formatCurrency
import java.text.NumberFormat
import java.util.Locale

@Composable
fun OverviewTabView(viewModel: MolyConsoleViewModel) {
    val overviewSummary by viewModel.overviewSummary.collectAsStateWithLifecycle()
    val isLoading by viewModel.isOverviewLoading.collectAsStateWithLifecycle()
    val error by viewModel.overviewError.collectAsStateWithLifecycle()
    
    when {
        isLoading -> {
            Box(
                modifier = Modifier.fillMaxSize(),
                contentAlignment = Alignment.Center
            ) {
                CircularProgressIndicator(color = Gold)
            }
        }
        error != null -> {
            val errorMsg = error
            OverviewErrorState(error = errorMsg ?: "Unknown error") {
                viewModel.loadOverview()
            }
        }
        overviewSummary != null -> {
            val summary = overviewSummary
            if (summary != null) {
                OverviewContent(summary = summary)
            } else {
                OverviewEmptyState()
            }
        }
        else -> {
            OverviewEmptyState()
        }
    }
}

@Composable
fun OverviewContent(summary: com.example.apk.console.models.OverviewSummary) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(20.dp)
    ) {
        Text(
            text = "Quick Overview",
            fontSize = 20.sp,
            fontWeight = FontWeight.Bold,
            color = TextPrimary,
            modifier = Modifier.fillMaxWidth()
        )
        
        // Summary Cards Grid
        LazyVerticalGrid(
            columns = GridCells.Fixed(2),
            horizontalArrangement = Arrangement.spacedBy(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp),
            modifier = Modifier.height(200.dp)
        ) {
            items(listOf(
                SummaryCardData(
                    title = "Total Balance",
                    value = formatCurrency(summary.totalBalance),
                    color = Color(0xFF4CAF50),
                    icon = Icons.Default.AccountBalance
                ),
                SummaryCardData(
                    title = "This Month",
                    value = formatCurrency(summary.thisMonthSpending),
                    color = Color(0xFFE53935),
                    icon = Icons.Default.CalendarToday
                ),
                SummaryCardData(
                    title = "Savings Rate",
                    value = "${String.format(Locale.getDefault(), "%.1f", summary.savingsRate)}%",
                    color = Color(0xFF4CAF50),
                    icon = Icons.Default.TrendingUp
                ),
                SummaryCardData(
                    title = "Active Goals",
                    value = "${summary.activeGoalsCount}",
                    color = Gold,
                    icon = Icons.Default.TrackChanges
                )
            )) { data ->
                SummaryCard(data = data)
            }
        }
        
        // AI Insight Card
        summary.latestInsight?.let { insight ->
            AIInsightCard(insight = insight)
        }
    }
}

data class SummaryCardData(
    val title: String,
    val value: String,
    val color: Color,
    val icon: androidx.compose.ui.graphics.vector.ImageVector
)

@Composable
fun SummaryCard(data: SummaryCardData) {
    GlassCard(
        padding = PaddingValues(16.dp),
        cornerRadius = 16.dp
    ) {
        Column(
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Row {
                Icon(
                    imageVector = data.icon,
                    contentDescription = null,
                    tint = data.color,
                    modifier = Modifier.size(20.dp)
                )
                Spacer(modifier = Modifier.weight(1f))
            }
            
            Text(
                text = data.title,
                fontSize = 12.sp,
                fontWeight = FontWeight.Medium,
                color = TextSecondary
            )
            
            Text(
                text = data.value,
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = TextPrimary
            )
        }
    }
}

@Composable
fun AIInsightCard(insight: com.example.apk.console.models.AIInsight) {
    GlassCard(
        padding = PaddingValues(20.dp),
        cornerRadius = 16.dp
    ) {
        Column(
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Icon(
                    imageVector = getInsightIcon(insight.type),
                    contentDescription = null,
                    tint = getInsightColor(insight.type),
                    modifier = Modifier.size(20.dp)
                )
                Text(
                    text = "AI Insight",
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = TextSecondary
                )
            }
            
            Text(
                text = insight.message,
                fontSize = 15.sp,
                color = TextPrimary,
                maxLines = 3
            )
        }
    }
}

@Composable
private fun OverviewErrorState(error: String, onRetry: () -> Unit) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(60.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Icon(
            imageVector = Icons.Default.Error,
            contentDescription = null,
            tint = Color(0xFFE53935).copy(alpha = 0.7f),
            modifier = Modifier.size(48.dp)
        )
        
        Text(
            text = "Unable to Load Overview",
            fontSize = 18.sp,
            fontWeight = FontWeight.Bold,
            color = TextPrimary
        )
        
        Text(
            text = error,
            fontSize = 14.sp,
            color = TextSecondary,
            modifier = Modifier.padding(horizontal = 32.dp)
        )
        
        Button(
            onClick = onRetry,
            colors = ButtonDefaults.buttonColors(containerColor = Gold)
        ) {
            Text("Retry", color = Charcoal)
        }
    }
}

@Composable
private fun OverviewEmptyState() {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(60.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Icon(
            imageVector = Icons.Default.BarChart,
            contentDescription = null,
            tint = TextSecondary.copy(alpha = 0.5f),
            modifier = Modifier.size(48.dp)
        )
        
        Text(
            text = "No Overview Data",
            fontSize = 18.sp,
            fontWeight = FontWeight.SemiBold,
            color = TextPrimary
        )
        
        Text(
            text = "Upload statements to see your financial overview",
            fontSize = 14.sp,
            color = TextSecondary,
            modifier = Modifier.padding(horizontal = 32.dp)
        )
    }
}


private fun getInsightIcon(type: com.example.apk.console.models.InsightType): androidx.compose.ui.graphics.vector.ImageVector {
    return when (type) {
        com.example.apk.console.models.InsightType.SPENDING_ALERT -> Icons.Default.Warning
        com.example.apk.console.models.InsightType.GOAL_PROGRESS -> Icons.Default.CheckCircle
        com.example.apk.console.models.InsightType.INVESTMENT_RECOMMENDATION -> Icons.Default.TrendingUp
        com.example.apk.console.models.InsightType.BUDGET_TIP -> Icons.Default.Lightbulb
        com.example.apk.console.models.InsightType.SAVINGS_OPPORTUNITY -> Icons.Default.AccountBalance
    }
}

private fun getInsightColor(type: com.example.apk.console.models.InsightType): Color {
    return when (type) {
        com.example.apk.console.models.InsightType.SPENDING_ALERT -> Color(0xFFFFEB3B)
        com.example.apk.console.models.InsightType.GOAL_PROGRESS -> Color(0xFF4CAF50)
        com.example.apk.console.models.InsightType.INVESTMENT_RECOMMENDATION -> Color(0xFF9C27B0)
        com.example.apk.console.models.InsightType.BUDGET_TIP -> Color(0xFF2196F3)
        com.example.apk.console.models.InsightType.SAVINGS_OPPORTUNITY -> Color(0xFF4CAF50)
    }
}

File: ios_monytix/ios_monytix/console/OverviewTabView.swift
//
//  OverviewTabView.swift
//  ios_monytix
//
//  Created by santosh on 05/01/26.
//

import SwiftUI

struct OverviewTabView: View {
    @ObservedObject var viewModel: MolyConsoleViewModel
    
    private let goldColor = Color(red: 0.831, green: 0.686, blue: 0.216) // #D4AF37
    private let charcoalColor = Color(red: 0.18, green: 0.18, blue: 0.18) // #2E2E2E
    
    var body: some View {
        ScrollView {
            VStack(spacing: 20) {
                if viewModel.isOverviewLoading {
                    ProgressView()
                        .tint(goldColor)
                        .padding(.top, 40)
                } else if let error = viewModel.overviewError {
                    errorState(error)
                } else if let summary = viewModel.overviewSummary {
                    overviewContent(summary)
                } else {
                    emptyState
                }
            }
            .padding(20)
        }
        .refreshable {
            await viewModel.loadOverview()
        }
    }
    
    private func overviewContent(_ summary: OverviewSummary) -> some View {
        VStack(spacing: 20) {
            // Quick Overview Section
            Text("Quick Overview")
                .font(.system(size: 20, weight: .bold))
                .foregroundColor(.white)
                .frame(maxWidth: .infinity, alignment: .leading)
            
            // Summary Cards Grid
            LazyVGrid(columns: [
                GridItem(.flexible(), spacing: 16),
                GridItem(.flexible(), spacing: 16)
            ], spacing: 16) {
                // Total Balance
                SummaryCard(
                    title: "Total Balance",
                    value: formatCurrency(summary.totalBalance),
                    color: .green,
                    icon: "dollarsign.circle.fill"
                )
                
                // This Month
                SummaryCard(
                    title: "This Month",
                    value: formatCurrency(summary.thisMonthSpending),
                    color: .red,
                    icon: "calendar"
                )
                
                // Savings Rate
                SummaryCard(
                    title: "Savings Rate",
                    value: "\(String(format: "%.1f", summary.savingsRate))%",
                    color: .green,
                    icon: "chart.line.uptrend.xyaxis"
                )
                
                // Active Goals
                SummaryCard(
                    title: "Active Goals",
                    value: "\(summary.activeGoalsCount)",
                    color: goldColor,
                    icon: "target"
                )
            }
            
            // AI Insight Card
            if let insight = summary.latestInsight {
                AIInsightCard(insight: insight)
            }
        }
    }
    
    private func errorState(_ error: String) -> some View {
        VStack(spacing: 16) {
            Image(systemName: "exclamationmark.triangle.fill")
                .font(.system(size: 48))
                .foregroundColor(.red.opacity(0.7))
            
            Text("Unable to Load Overview")
                .font(.system(size: 18, weight: .bold))
                .foregroundColor(.white)
            
            Text(error)
                .font(.system(size: 14))
                .foregroundColor(.gray.opacity(0.7))
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
            
            Button("Retry") {
                Task {
                    await viewModel.loadOverview()
                }
            }
            .buttonStyle(GoldButtonStyle())
            .padding(.top, 8)
        }
        .padding(.top, 60)
    }
    
    private var emptyState: some View {
        VStack(spacing: 16) {
            Image(systemName: "chart.bar.doc.horizontal")
                .font(.system(size: 48))
                .foregroundColor(.gray.opacity(0.5))
            
            Text("No Overview Data")
                .font(.system(size: 18, weight: .semibold))
                .foregroundColor(.white)
            
            Text("Upload statements to see your financial overview")
                .font(.system(size: 14))
                .foregroundColor(.gray.opacity(0.7))
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
        }
        .padding(.top, 60)
    }
    
    private func formatCurrency(_ amount: Double) -> String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .currency
        formatter.currencySymbol = "₹"
        formatter.maximumFractionDigits = 0
        return formatter.string(from: NSNumber(value: amount)) ?? "₹0"
    }
}

// MARK: - Summary Card

struct SummaryCard: View {
    let title: String
    let value: String
    let color: Color
    let icon: String
    
    var body: some View {
        GlassCard(padding: 16, cornerRadius: 16) {
            VStack(alignment: .leading, spacing: 12) {
                HStack {
                    Image(systemName: icon)
                        .font(.system(size: 20))
                        .foregroundColor(color)
                    
                    Spacer()
                }
                
                Text(title)
                    .font(.system(size: 12, weight: .medium))
                    .foregroundColor(.gray.opacity(0.7))
                
                Text(value)
                    .font(.system(size: 20, weight: .bold))
                    .foregroundColor(.white)
            }
        }
    }
}

// MARK: - AI Insight Card

struct AIInsightCard: View {
    let insight: AIInsight
    
    private let goldColor = Color(red: 0.831, green: 0.686, blue: 0.216)
    
    var body: some View {
        GlassCard(padding: 20, cornerRadius: 16) {
            VStack(alignment: .leading, spacing: 12) {
                HStack {
                    Image(systemName: insight.type.icon)
                        .font(.system(size: 20))
                        .foregroundColor(colorForType(insight.type))
                    
                    Text("AI Insight")
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundColor(.gray.opacity(0.7))
                    
                    Spacer()
                }
                
                Text(insight.message)
                    .font(.system(size: 15))
                    .foregroundColor(.white)
                    .lineLimit(3)
            }
        }
    }
    
    private func colorForType(_ type: InsightType) -> Color {
        switch type {
        case .spendingAlert:
            return .yellow
        case .goalProgress:
            return .green
        case .investmentRecommendation:
            return .purple
        case .budgetTip:
            return .blue
        case .savingsOpportunity:
            return .green
        }
    }
}

#Preview {
    OverviewTabView(viewModel: MolyConsoleViewModel(authService: AuthService()))
        .background(Color(red: 0.18, green: 0.18, blue: 0.18))
}

